image_view/src/extract_images.cpp

```diff
@@ -34,121 +34,137 @@
 
 #include <opencv2/highgui/highgui.hpp>
 
-#include <ros/ros.h>
-#include <sensor_msgs/Image.h>
+#include <rclcpp/rclcpp.hpp>
 #include <cv_bridge/cv_bridge.h>
 #include <image_transport/image_transport.h>
+#include <sensor_msgs/msg/image.hpp>
 
-#include <boost/thread.hpp>
 #include <boost/format.hpp>
 
-class ExtractImages
+#include <chrono>
+#include <mutex>
+#include <thread>
+#include <string>
+
+namespace image_view
+{
+
+class ExtractImagesNode
+  : public rclcpp::Node
 {
 private:
   image_transport::Subscriber sub_;
 
-  sensor_msgs::ImageConstPtr last_msg_;
-  boost::mutex image_mutex_;
+  sensor_msgs::msg::Image::ConstSharedPtr last_msg_;
+  std::mutex image_mutex_;
 
   std::string window_name_;
   boost::format filename_format_;
   int count_;
-  double _time;
+  rclcpp::Time _time;
   double sec_per_frame_;
 
-#if defined(_VIDEO)
+#ifdef _VIDEO
+
   CvVideoWriter* video_writer;
-#endif //_VIDEO
+
+#endif  // _VIDEO
 
 public:
-  ExtractImages(const ros::NodeHandle& nh, const std::string& transport)
-    : filename_format_(""), count_(0), _time(ros::Time::now().toSec())
+  ExtractImagesNode(const rclcpp::NodeOptions & options, const std::string & transport)
+    : rclcpp::Node("extract_images_node", options),
+      filename_format_(""), count_(0), _time(this->now())
   {
-    std::string topic = nh.resolveName("image");
-    ros::NodeHandle local_nh("~");
+    auto topic = rclcpp::expand_topic_or_service_name("image", this->get_name(), this->get_namespace());
 
-    std::string format_string;
-    local_nh.param("filename_format", format_string, std::string("frame%04i.jpg"));
+    if (topic == "/image") {
+      RCLCPP_WARN(
```

Comment:
I think this is a debug

Reply:
I don't think it should be. It's to let the user know that they may have intended to remap the `image` topic but they didn't. Either way, this works properly now.